simulator/syncer/syncer.go

```diff
@@ -0,0 +1,339 @@
+package syncer
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+// GVRs is a list of GroupVersionResource that we want to sync.
+// Note that this order matters - When first importing resources, we want to sync namespaces first, then priorityclasses, storageclasses...
+var GVRs = []schema.GroupVersionResource{
+	{Group: "", Version: "v1", Resource: "namespaces"},
+	{Group: "scheduling.k8s.io", Version: "v1", Resource: "priorityclasses"},
+	{Group: "storage.k8s.io", Version: "v1", Resource: "storageclasses"},
+	{Group: "", Version: "v1", Resource: "persistentvolumeclaims"},
+	{Group: "", Version: "v1", Resource: "nodes"},
+	// Note that we are not syncing pods that are already scheduled.
+	{Group: "", Version: "v1", Resource: "pods"},
+	{Group: "", Version: "v1", Resource: "persistentvolumes"},
+}
+
+// Event is a type of events that occur in the source cluster.
+type Event int
+
+const (
+	Add Event = iota
+	Update
+)
+
+// MutatingFunctions is a map of GroupVersionResource to mutatingFunction.
+// This is public so that outside users can add their own mutating functions.
+var MutatingFunctions = map[schema.GroupVersionResource]MutatingFunction{
+	{Group: "", Version: "v1", Resource: "persistentvolumes"}: mutatePV,
+}
```

Comment:
Modifying global variables are generally dangerous.
How about

1. Defining `defaultMutatingFunctions()` that returns the default list of mutating functions, and
2. Adding a parameter to `New()` so that users can pass a list